### ISSUE TITLE
Can't use Aloha inside a form

### ISSUE THREAD

**Issue by Pajamaman** [alohaeditor/Aloha-Editor#1335]
First of all, I can't find an example that demonstrates how to use Aloha inside a form. So it's possible that I'm doing it wrong.

My form looks like this:

    <form action="foo.php">
        <div class="editor-commands">
            <button class="editor-command-bold">B</button>
        </div>
        <div class="editor">
            <p>Blah blah blah...</p>
        </div>
    </form>

Because the bold button is inside the form, clicking on it submits the form. Normally, this would not be a problem. But the Aloha commands only work if the click event handler is defined exactly as follows:

    document.querySelector('.editor-command-bold').addEventListener('click', aloha.command(aloha.commands.bold));

This doesn't work at all:

    document.querySelector('.editor-command-bold').addEventListener('click', function (event) {
        event.preventDefault();
        aloha.command(aloha.commands.bold);
    });

Nor does this:

    document.querySelector('.editor-command-bold').addEventListener('click', function (event) {
        event.preventDefault();
        aloha.command(aloha.commands.bold)(event);
    });

I also tried using span elements instead of buttons, but this doesn't work either:

    <form action="foo.php">
        <div class="editor-commands">
            <span class="editor-command-bold">B</span>
        </div>
        <div class="editor">
            <p>Blah blah blah...</p>
        </div>
    </form>

For some reason, the Aloha commands only work on button elements.

The only acceptable solution I was able to come up with was to define a second event handler like so:

    document.querySelector('.editor-command-bold').addEventListener('click', aloha.command(aloha.commands.bold));
    document.querySelector('.editor-command-bold').addEventListener('click', function (event) {
        event.preventDefault();
    });

This seems pretty clunky.

In conclusion, if you want to use Aloha inside a form, you have to place the controls outside the form using button elements, or define separate event handlers for the controls to prevent the form from submitting. Is this by design or am I missing something?

**Comment by petrosalema** [comment]
Your second approach was in the right direction, but incomplete.
```aloha.ui.command``` generates a function that must be invoked (www.alohaeditor.org/api/ui.html#command).

I would recommend that you don't generate this function on each click event but one time like this:

```javascript
var makeBold =  aloha.command(aloha.commands.bold);
document.querySelector('.editor-command-bold').addEventListener('click', function (event) {
    event.preventDefault();
   makeBold(event);
});
```
